Jenkinsfile

```diff
@@ -432,8 +432,16 @@ pipeline {
         ansiColor('xterm')
     }
 
+    // Default priority is 2, lower is better.
+    // The parameter for a job is set using the script/Jenkinsfile from the
+    // previous build, so the first build of any PR will always run at
+    // priority 2, but subsequent ones will use the value from here.  The
+    // advantage therefore is not to change the priority of PRs, but to
+    // change the master branch itself.
     parameters {
-        string(name: 'BuildPriority', defaultValue: '', description: 'Priority of the build.  DO NOT USE WITHOUT PERMISSION.')
+        string(name: 'BuildPriority',
+	       defaultValue: env.BRANCH_NAME == 'master' ? '3' : '2',
```

Comment:
I would prefer moving the logic into a function that can be called here, like https://github.com/daos-stack/daos/blob/e47cea0f6b714a464f9378787197d0b2bee76736/Jenkinsfile#L446 and https://github.com/daos-stack/daos/blob/e47cea0f6b714a464f9378787197d0b2bee76736/Jenkinsfile#L414-L422 to keep the amount of byte-code inside the `pipeline {}` block to a bare minimum.

You wouldn't need the `node {}` but just move any logic into a function.  I would also prefer we keep the non-master priority to `''` and set master to default+1, but I'm not sure if we can query for the default, so setting it to `4` here for the moment would be fine.

Reply:
I did try many iterations of this earlier but was missing the node/checkoutScm parts so couldn't get it to work.  Agreed that we could keep the non-master priority to '' for clarity but in practice it'll be the same as you say.  I even had a 'Jump-queue: true' commit-pragma in an earlier version.